exporter/exporterhelper/internal/queuebatch/queue_batch.go

```diff
@@ -63,8 +63,6 @@ func newQueueBatch(
 
 	var b Batcher[request.Request]
 	if cfg.Batch != nil {
-		// TODO: https://github.com/open-telemetry/opentelemetry-collector/issues/12244
-		cfg.NumConsumers = 1
```

Comment:
Please move this to line 83

Reply:
And link the sharding issue, since without that setting this to any other number is a waste of resource.